R/LoadingSaving.R

```diff
@@ -211,27 +219,64 @@ getAndromedaTempDiskSpace <- function(andromeda = NULL) {
   if (!is.null(andromeda) && !inherits(andromeda, "SQLiteConnection")) 
     stop("Andromeda argument must be of type 'Andromeda'.")
   
-  # Using Java because no cross-platform functions available in R:
-  if (!.isInstalled("rJava")) {
-    return(NA)
+  if (is.null(andromeda)) {
+    folder <- .getAndromedaTempFolder()
   } else {
-    if (is.null(andromeda)) {
-      folder <- .getAndromedaTempFolder()
+    folder <- dirname(andromeda@dbname) 
+  }
+  
+  if(!dir.exists(folder)) {
+    msg <- paste("Andromeda temp folder does not exist\n",
+                 "folder:", folder, "\n",
+                 "Cannot determine available Andromeda disk space.")
+    rlang::warn(msg, class = "Andromeda")
+    return(NA)
+  }
+  
+  osInfo <- tolower(paste(sessionInfo()$platform, sessionInfo()$running))
+  os <- regmatches(osInfo, regexpr("windows|mac|linux", osInfo))
+  if(!(os %in% c("windows", "mac", "linux"))) {
+    msg <- "Operating system cannot be determined.\nCannot get available Andromeda disk space.\n"
+    rlang::warn(msg, class = "Andromeda")
+    return(NA)
+  }
+  
+  if (os == "windows") {
+    # https://stackoverflow.com/questions/32200879/how-to-get-disk-space-of-windows-machine-with-r
+    # https://stackoverflow.com/questions/293780/free-space-in-a-cmd-shell
+    disks <- system("wmic logicaldisk get size,freespace,caption", inter = T)
+    disks <- read.table(textConnection(disks), 
+                        skip = 1,
+                        col.names =  c("disk", "freeSpace", "size"),
+                        row.names = NULL)
+    
+    idRow <- stringr::str_detect(folder, paste0("^",disks$disk))
```

Comment:
Would you mind adding stringr to the DESCRIPTION Imports? 

Would we need to make this case-insensitive?